op-deployer/pkg/deployer/verify/verifier.go

```
@@ -88,108 +56,61 @@ func VerifyCLI(cliCtx *cli.Context) error {
-				v.numFailed++
-				v.log.Error("failed to verify contract", "name", contractName, "error", err)
-			}
+		err := v.VerifyContract(ctx, addr, contractName)
+		if err == nil {
+			numVerified++
+		} else if err == ErrAlreadyVerified {
+			numSkipped++
+		} else {
+			return fmt.Errorf("failed to verify contract %s: %w", contractName, err)
 		}
-	}
-	return nil
-}
-
-func (v *Verifier) verifySingleContract(ctx context.Context, address common.Address, contractName string) error {
-	verified, err := v.etherscan.isVerified(address)
-	if err != nil {
-		return fmt.Errorf("failed to check verification status: %w", err)
-	}
-	if verified {
-		v.log.Info("Contract is already verified", "name", contractName, "address", address.Hex())
-		v.numSkipped++
-		return nil
-	}
-
-	v.log.Info("Formatting etherscan verify request", "name", contractName, "address", address.Hex())
-	artifact, err := v.getContractArtifact(contractName)
-	if err != nil {
-		return fmt.Errorf("failed to get contract source: %w", err)
-	}
-
-	constructorArgs, err := v.getConstructorArgs(ctx, address, artifact)
-	if err != nil {
-		return fmt.Errorf("failed to get constructor args: %w", err)
-	}
-
-	reqId, err := v.etherscan.verifySourceCode(address, artifact, constructorArgs)
-	if err != nil {
-		return fmt.Errorf("failed to verify contract: %w", err)
-	}
-	v.log.Info("Verification request submitted", "name", contractName, "address", address.Hex())
-
-	if err = v.etherscan.pollVerificationStatus(reqId); err != nil {
-		return fmt.Errorf("failed when checking verification status: %w", err)
+	} else {
+		numVerified, numSkipped, numFailed = v.VerifyContracts(ctx, bundle)
 	}
 
-	v.log.Info("Verification complete", "name", contractName, "address", address.Hex())
-	v.numVerified++
+	l.Info("--- COMPLETE ---")
+	l.Info("final results", "numVerified", numVerified, "numSkipped", numSkipped, "numFailed", numFailed)
+	// May want to return an error here if numFailed > 0
```

Agreed this should return an error if numFailed > 0.